src/test/kotlin/no/nav/tilleggsstonader/sak/vedtak/læremidler/beregning/LæremidlerVedtaksperiodeUtilTest.kt

```diff
@@ -208,4 +212,196 @@ class LæremidlerVedtaksperiodeUtilTest {
+        val vedtaksperioderJanMars = listOf(vedtaksperiodeJanFeb, vedtaksperiodeMars)
+        val førsteMars: LocalDate = LocalDate.of(2025, 3, 1)
+        val femtendeMars: LocalDate = LocalDate.of(2025, 3, 15)
+        val førsteApril: LocalDate = LocalDate.of(2025, 4, 1)
+
+        @Test
+        fun `kaster ikke feil ved ingen revurder fra og ingen gamle perioder (førstegangsbehandling)`() {
+            assertDoesNotThrow {
+                validerIngenEndringerFørRevurderFra(
+                    vedtaksperioder = vedtaksperioderJanMars,
+                    vedtaksperioderForrigeBehandling = emptyList(),
+                    revurderFra = null,
+                )
+            }
+        }
+
+        @Nested
+        inner class NyPeriode {
+            @Test
+            fun `kaster ikke feil ved ny periode som starter etter revurder fra`() {
+                assertDoesNotThrow {
+                    validerIngenEndringerFørRevurderFra(
+                        vedtaksperioder = vedtaksperioderJanMars,
+                        vedtaksperioderForrigeBehandling = vedtaksperioderJanFeb,
+                        revurderFra = førsteMars,
+                    )
+                }
+            }
+
+            @Test
+            fun `kaster feil ved ny periode med fom før revurder fra`() {
+                assertThrows<ApiFeil> {
+                    validerIngenEndringerFørRevurderFra(
+                        vedtaksperioder = vedtaksperioderJanMars,
+                        vedtaksperioderForrigeBehandling = vedtaksperioderJanFeb,
+                        revurderFra = femtendeMars,
+                    )
+                }
+            }
+
+            @Test
+            fun `kaster feil ved ny periode med fom og tom før revuder fra`() {
+                assertThrows<ApiFeil> {
+                    validerIngenEndringerFørRevurderFra(
+                        vedtaksperioder = vedtaksperioderJanMars,
+                        vedtaksperioderForrigeBehandling = vedtaksperioderJanFeb,
+                        revurderFra = førsteApril,
+                    )
+                }
+            }
+
+            @Test
+            fun `kaster feil ved ny periode som er lik eksisterende periode lagt til før revuder fra`() {
+                val nyeVedtaksperioder =
+                    listOf(
+                        vedtaksperiodeJanFeb,
+                        vedtaksperiodeJanFeb,
+                    )
+
+                assertThrows<ApiFeil> {
+                    validerIngenEndringerFørRevurderFra(
+                        vedtaksperioder = nyeVedtaksperioder,
+                        vedtaksperioderForrigeBehandling = vedtaksperioderJanFeb,
+                        revurderFra = førsteMars,
+                    )
+                }
+            }
+
+            @Test
+            fun `kaster feil ved nye perioder før revurder fra etter opphør med ingen eksisterende vedtaksperioder`() {
+                assertThrows<ApiFeil> {
```

Comment:
Hva med å sjekke på teksten i feilmeldingen? eg:` Det er ikke tillat å endre perioder fra før revurder fra dato`
dette for å være tydlig på hvilken feil testen skal trigge